src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -193,6 +198,64 @@ public void deleteRoles(final Node node) throws RepositoryException {
     };
 
     /**
+     *  This maps a Collection of acl:agentClass values to a List of agents.
+     *  Any out-of-domain URIs are silently ignored.
+     */
+    private List<String> dereferenceAgentClasses(final Collection<String> agentClasses) {
+        final Session internalSession = sessionFactory.getInternalSession();
+        final IdentifierConverter<Resource, FedoraResource> translator =
+                new DefaultIdentifierTranslator(internalSession);
+
+        final List<String> members = new ArrayList<>();
+        agentClasses.stream()
+                    .distinct()
+                    .forEach(x -> {
+                        if (x.startsWith(FEDORA_INTERNAL_PREFIX)) {
+                            final FedoraResource resource = nodeService.find(
```

Comment:
Should we be testing/requiring that the `FedoraResource` is of `rdf:type`= `foaf:Group`?
It is not clear from the spec that this is a requirement... but you mention it in the associated JIRA ticket and include `foaf:Group` in the integration test example.